test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,317 @@
+package docker_e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"celestiaorg/celestia-app/test/docker-e2e/dockerchain"
+
+	"github.com/celestiaorg/tastora/framework/testutil/wait"
+	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+)
+
+// TallyResponse represents the JSON response from the signal tally query
+// We had to define it again because the JSON response type is not a number but string
+type TallyResponse struct {
+	VotingPower      string `json:"voting_power"`
+	ThresholdPower   string `json:"threshold_power"`
+	TotalVotingPower string `json:"total_voting_power"`
+}
+
+// UpgradeConfig holds configuration parameters for performing an upgrade test
+// on the Celestia chain.
+type UpgradeConfig struct {
+	// BaseBinaryVersion is the initial version of the chain before upgrade.
+	BaseBinaryVersion string
+	// TargetBinaryVersion is the version to which the chain will be upgraded.
+	TargetBinaryVersion string
+	// TargetAppVersion is the app version to upgrade to (used in major upgrades).
+	TargetAppVersion uint64
+}
```

Comment:
I don't understand why there need to be two separate binary versions for the major upgrade test. The test could just use the v4.0.10 multiplexer image. IMO it doesn't make sense to re-use this `UpgradeConfig` for the Minor and Major test because the minor test uses two different binary versions but the major test uses two different app versions.